src/vikcoord.c

```diff
@@ -225,3 +225,81 @@ gdouble vik_coord_angle (const VikCoord *vc1, const VikCoord *vc2)
   angle = fmod (RAD2DEG(angle) + 360.0, 360);
   return angle;
 }
+
+/**
+ * vik_coord_angle_end:
+ *
+ * Get angle of first coord from the second one in degrees (final bearing). Basically the inverse of vik_coord_angle.
+ *
+ */
+gdouble vik_coord_angle_end (const VikCoord *vc1, const VikCoord *vc2)
+{
+  struct LatLon ll1, ll2;
+  vik_coord_to_latlon ( vc1, &ll1 );
+  vik_coord_to_latlon ( vc2, &ll2 );
+
+  // Convert to radians for use in the algorithm
+  gdouble rlat1 = DEG2RAD(ll1.lat);
+  gdouble rlong1 = DEG2RAD(ll1.lon);
+  gdouble rlat2 = DEG2RAD(ll2.lat);
+  gdouble rlong2 = DEG2RAD(ll2.lon);
+
+  // This formula is for the final bearing
+  //θ = atan2( sin Δλ ⋅ cos φ2 , cos φ1 ⋅ sin φ2 − sin φ1 ⋅ cos φ2 ⋅ cos Δλ )
+  // in -pi to +pi radians
+  gdouble dLon = (rlong2 - rlong1);
+  gdouble y = sin(dLon) * cos(rlat1);
+  gdouble x = (-cos(rlat2) * sin(rlat1)) + (sin(rlat2) * cos(rlat1) * cos(dLon));
+  gdouble angle = atan2(y, x);
+
+  // Bring into range 0..360 degrees
+  angle = fmod (RAD2DEG(angle) + 360.0, 360);
+  return angle;
+}
+
+/**
+ * vik_coord_geodesic_waypoint:
+ *
+ * Calculate the waypoint at angular distance ratio n between vc1 and vc2. n = 0 returns vc1, n = 1 returns vc2.
+ * Used for drawing great circles for the ruler tool.
+ * For the relevant formulas, refer to https://en.wikipedia.org/wiki/Great-circle_navigation
+ *
+ */
+void vik_coord_geodesic_waypoint (const VikCoord *vc1, const VikCoord *vc2, gdouble n, VikCoord *wpt)
```

Comment:
I think it would be best to avoid calling this returned point 'a waypoint' (and hence the function name) as it will be confused with actual user Waypoints (i.e. the VikWaypoint type).
Perhaps call it vik_coord_geodesic_coord() and change wpt --> rvc, as in Result VikCoord.
& "Calculate the waypoint at angular distance..." --> "Calculate the coordinate at angular distance..."